### ISSUE TITLE
minimum_trials breaks counterbalance in some cases

### ISSUE THREAD

**Issue by HuangHam** [sweetpea-org/sweetpea-py#47]
Loving Sweatpea! Have an issue tho. Consider the minimal Stroop example below:
```
correct_response = Factor(name="correct_response", initial_levels=["H", "S"])
congruency = Factor(name="congruency", initial_levels=["congruent", "incongruent"])
stroop_design     = [correct_response, congruency]
stroop_crossing   = [correct_response, congruency]
```
Obviously after crossing there are 4 unique trial conditions. The goal is that we want to have the same amount of trials in each condition if the total trial number is a multiple of 4. If I do the following, it works as expected:
```
stroop_block  = fully_cross_block(design=stroop_design, crossing=stroop_crossing, constraints=[MinimumTrials(trials=4)])
stroop_experiments = synthesize_trials_non_uniform(block=stroop_block, samples=1)
tabulate_experiments(experiments=stroop_experiments, factors=stroop_crossing)
```
The following also works:
```
stroop_block  = fully_cross_block(design=stroop_design, crossing=stroop_crossing, constraints=[])
stroop_experiments = synthesize_trials_non_uniform(block=stroop_block, samples=1)
tabulate_experiments(experiments=stroop_experiments, factors=stroop_crossing)
```
However, if I change MinimumTrials to 8, it does not generate the same amount of trials in each condition:
```
stroop_block  = fully_cross_block(design=stroop_design, crossing=stroop_crossing, constraints=[MinimumTrials(trials=8)])
stroop_experiments = synthesize_trials_non_uniform(block=stroop_block, samples=1)
tabulate_experiments(experiments=stroop_experiments, factors=stroop_crossing)
```
Output:
```
correct_response H | congruency congruent   | frequency 1 | proportion 12.5%
correct_response H | congruency incongruent | frequency 1 | proportion 12.5%
correct_response S | congruency congruent   | frequency 2 | proportion 25.0%
correct_response S | congruency incongruent | frequency 4 | proportion 50.0%
```
Another feature we would like is when the minimumtrials is not a multiple of the number of conditions, we would still want each trial to show up as evenly as possible and randomly assign conditions for the reminder trials. Right now it's now the case either.

**Comment by mflatt** [comment]
This problem is not yet fixed, although I've fixed it for the specific example (in GitHub, not yet pushed to pip).

The example now works because I extended the combinatoric solver to support `minimum_trials`. I still need to fix the solver that was previously triggered by the use of `minimum_trials`.

**Comment by mflatt** [comment]
The problem should be fixed the right way with f54eb52f6d4549dfc15fca84e42174c591a00158, so that the example also works with `synthesize_trials` and `sampling_strategy = NonUniformSamplingStrategy`.
